### ISSUE TITLE
Open OnDemand UI Issue: Newly Mounted AWS S3 Bucket Not Visible in Files Section

### ISSUE THREAD

**Issue by mharini** [OSC/ondemand#3733]
Hi,

We are currently encountering an issue with OnDemand when attempting to view a mounted S3 bucket in the UI. A new bucket was created with the appropriate permissions and added to the user groups. We run a cron job to mount/unmount the S3 buckets in periodic basis, based on user group and permissions. The bucket appears in the mounted path, and we can list the files from the cluster shell. However, they are not accessible from the OnDemand UI. While restarting the OnDemand service resolved the issue, we are seeking a solution that doesn't require a restart.

Any inputs would be great to resolve this issue.
<img width="481" alt="Screenshot 2024-08-16 at 3 40 31 PM" src="https://github.com/user-attachments/assets/f7f95cb6-46d5-4f22-81a3-a03643ad3832">

**Comment by johrstrom** [comment]
> While restarting the OnDemand service resolved the issue, we are seeking a solution that doesn't require a restart.

I'd guess it's an issue of when and how it's mounted. If it's mounted as a filesystem just like any other NFS drive that's been mounted I can only think there's a race condition of some kind. Like we're trying to access it when it's not available yet.

How and when is it mounted?

**Comment by mharini** [comment]
The S3 buckets are mounted using s3fs using a cron job running on a periodic basis, to identify if any new buckets or permissions or groups are added/modified. 

We also see that "Restart Web Server" in the help option fixes the issue without restarting the all of the ondemand.

**Comment by mharini** [comment]
We have modified the cron job to kill the nginx process for a specific user using pid whenever there is a change in group or user.

We also identified comments in the existing code base that the web server should be restarted automatically when there is a change in the group or user. Any additional configurations to be done for this to happen?

**Comment by johrstrom** [comment]
> Any additional configurations to be done for this to happen?

I don't think so. IIRC the app checks to see if any groups have changed then provides a button for the user to restart? But that's just from memory.

**Comment by johrstrom** [comment]
Yea it seems to me that we could make some sort of notification under the condition that `Proccess::Groups` don't match `CurrentUser.groups`. Seems like we'd need to cache this result every 5 minutes or so, but in any case, when this cache is busted we could provide a notification that shows a notification that you likely need to restart the server (with a link provided).

**Comment by moffatsadeghi** [comment]
Review Meeting Details
Explained the issue is related to user groups being updated after a PUN (Per-User NGINX) has started
Clarified that OS-level changes affecting users require a PUN restart
Suggested detecting the change and either showing a warning or kicking users out
Conclusion
    The team decided to schedule this for version 4.2
    The proposed solution is to implement a notification system to inform users when they need to restart their PUN
